.github/workflows/holochain-build-and-test.yml

```diff
@@ -105,16 +105,22 @@ jobs:
 
           git status
 
-          # if a nix store path was passed, ensure it exists locally
+          # If a nix store path was passed, ensure it exists locally. This will fetch the path from cachix.
           if [[ ${{ inputs.repo_path }} == "/nix/"* ]]; then
             nix-store --realise ${{inputs.repo_path }}
+            
+            # Temporary workaround for https://github.com/NixOS/nix/issues/10202
+            git config --global --add safe.directory ${{ inputs.repo_path }}
           fi
 
           # first build all derivations. this could be enough for test derivations.
           nix build -L --show-trace \
```

Comment:
Moving to Nix 2.20 here -> https://github.com/holochain/holochain-infra/pull/92

Has caused an issue with Git (libgit2 embedded in Nix) rejecting files owned by `root` as in input -> https://github.com/holochain/holochain/actions/runs/9191914450/job/25280481018#step:4:90